WebUI/src/assets/js/store/backendServices.ts

```diff
@@ -0,0 +1,41 @@
+import { defineStore } from "pinia";
+import { useGlobalSetup } from "./globalSetup";
+
+export const useBackendServices = defineStore("backendServices", () => {
+    const globalSetup = useGlobalSetup();
+
+    const info = ref<ApiServiceInformation[]>([]);
+    window.electronAPI.getServices().then(services => {
+        info.value = services;
+    });
+    window.electronAPI.onServiceInfoUpdate(updatedInfo => {
+        info.value = info.value.map(oldInfo => oldInfo.serviceName === updatedInfo.serviceName ? updatedInfo : oldInfo);
+    });
+
+    const allRequiredSetUp = computed(() => info.value.length > 0 && info.value.filter(s => s.isRequired).every(s => s.isSetUp));
+    const allRequiredRunning = computed(() => info.value.length > 0 && info.value.filter(s => s.isRequired).every(s => s.status === "running"));
+
+    watch(allRequiredRunning, (newVal) => {
+        if (newVal) {
+            globalSetup.loadingState = "running";
```

Comment:
in globalSetup.ts, there is also a watch on this var. Can't we get rid of this watch then or is it technically required?